test/Shared/UserManagerTestBase.cs

```diff
@@ -1165,19 +1174,19 @@ public Task<IdentityResult> ValidateAsync(UserManager<TUser> manager, TUser user
             var context = CreateTestContext();
             var manager = CreateManager(context);
             var roleManager = CreateRoleManager(context);
-            var role = CreateRole("addUserTest");
+            var roleName = "AddUserTest" + Guid.NewGuid().ToString();
```

Comment:
Sounds good to move this to the helper. If we need to reason about the generated value in a entity type-agnostic way, the helper can have output parameters for that.